### PR TITLE
MGMT-13001: Adding cost by machine type metric

### PR DESCRIPTION
**Pull request by danmanor**
Adding cost by machine type to `equinix` section in the report

image

### CODE REVIEW

File: src/jobsautoreport/models.py [hunk 1]
```diff
@@ -20,3 +20,13 @@ class JobState(Enum):
...
+
+
+class MachineType(Enum):
+    M3_SMALL_X86 = "m3.small.x86"
+    C3_MEDUIM_X86 = "c3.medium.x86"
+    A3_LARGE_X86 = "a3.large.x86"
+    C3_LARGE_ARM64 = "c3.large.arm64"
+    M3_LARGE_X86 = "m3.large.x86"
+    N3_XLARGE_X86 = "n3.xlarge.x86"
+    S3_XLARGE_X86 = "s3.xlarge.x86"
```
Comment (adriengentil) [inline src/jobsautoreport/models.py:32]: machine type shouldn't be hardcoded as we change them (and equinix change them) quite often

File: src/jobsautoreport/report.py [hunk 2]
```diff
@@ -4,13 +4,30 @@
...
+from jobsautoreport.models import JobState, JobType, MachineType, StepState
 from jobsautoreport.query import Querier
+from prowjobsscraper.equinix_usages import EquinixUsageEvent
 from prowjobsscraper.event import JobDetails
 
 logger = logging.getLogger(__name__)
 
 
+class MachineMetrics(BaseModel):
+    m3_small_x86: float = 0
```
Comment (adriengentil) [inline src/jobsautoreport/report.py:16]:
```
>>> m3_small_x86: float = 0
>>> type(m3_small_x86)
<class 'int'>
```
The type annotation in python is not evaluated by the interpreter, the value should be of explicitly of type float

**Review by adriengentil — CHANGES_REQUESTED** [review]

**Comment by osherdp** [comment]
a bit busy, uncc-ing for now sorry about it
/uncc

**Comment by rccrdpccl** [comment]
/lgtm
although I miss a ton of context :)

**Review by eliorerz — APPROVED** [review]

**Pull request merged**